server.js

```diff
@@ -127,47 +127,10 @@ app.use(session({
   secret: crypto.randomBytes(64).toString('hex')
 }))
 
-// add nunjucks function called 'checked' to populate radios and checkboxes,
-// needs to be here as it needs access to req.session and nunjucks environment
-var addCheckedFunction = function (app, nunjucksEnv) {
-  app.use(function (req, res, next) {
-    nunjucksEnv.addGlobal('checked', function (name, value) {
-      // check session data exists
-      if (req.session.data === undefined) {
-        return ''
-      }
-
-      var storedValue = req.session.data[name]
-
-      // check the requested data exists
-      if (storedValue === undefined) {
-        return ''
-      }
-
-      var checked = ''
-
-      // if data is an array, check it exists in the array
-      if (Array.isArray(storedValue)) {
-        if (storedValue.indexOf(value) !== -1) {
-          checked = 'checked'
-        }
-      } else {
-        // the data is just a simple value, check it matches
-        if (storedValue === value) {
-          checked = 'checked'
-        }
-      }
-      return checked
-    })
-
-    next()
-  })
-}
-
 if (useAutoStoreData === 'true') {
   app.use(utils.autoStoreData)
-  addCheckedFunction(app, nunjucksAppEnv)
-  addCheckedFunction(documentationApp, nunjucksDocumentationEnv)
+  utils.addCheckedFunction(app, nunjucksAppEnv)
```

Comment:
It would be nice to be consistent by moving the `app.use` here, but I'm not sure what we'd do about the fact that the environment would still have to be passed in. It just seems a bit strange that `addCheckedFunction` is responsible for binding itself to the app.

Reply:
Yes, had the same concerns when discussing this with @joelanman. I'll try and think of a better way of sharing the env while trying to clean up the code.

Reply:
I've tried the alternative approach here: https://github.com/alphagov/govuk_prototype_kit/pull/447